### PR TITLE
feat: Make Rollout History Collapsable

### PR DESCRIPTION
**Pull request by Mrod1598**
### Proposed Change

image

image

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by shazlehu** [comment]
Looks good, but the ticket says "By default, we should collapse the rollout history section with a caret while leaving the most recent / current entry visible."

File: ui/src/components/RolloutHistory/RolloutHistory.tsx [hunk 1]
```diff
@@ -54,30 +61,53 @@ export const RolloutHistory: React.FC<RolloutHistoryProps> = ({
...
+      </Collapse>
+      <Box sx={{ ml: 3, mr: 3 }}>
+        <Grid
+          container
+          spacing={10}
+          direction="column"
+          alignItems="center"
+          justifyContent="center"
+        >
+          <Grid item flex={1} xs={12} sm={12}>
```
Comment (shazlehu) [inline ui/src/components/RolloutHistory/RolloutHistory.tsx:103]:
```suggestion
    <Stack direction="row" justifyContent="center" alignItems="center">
```

File: ui/src/components/RolloutHistory/RolloutHistory.tsx [hunk 2]
```diff
@@ -54,30 +61,53 @@ export const RolloutHistory: React.FC<RolloutHistoryProps> = ({
...
+              onToggleExpanded={() => setExpanded((prev) => !prev)}
             />
-          )}
-        </Stack>
-      )}
-
-      {messages}
+          </Grid>
+        </Grid>
+      </Box>
```
Comment (shazlehu) [inline ui/src/components/RolloutHistory/RolloutHistory.tsx:110]:
```suggestion
          </Stack>
```

File: ui/src/components/RolloutHistory/RolloutHistory.tsx [hunk 3]
```diff
@@ -54,30 +61,53 @@ export const RolloutHistory: React.FC<RolloutHistoryProps> = ({
...
-              data-testid="circular-progress"
-              disableShrink
+      <Stack>
+        <Typography fontSize={18} fontWeight={600} marginBottom="8px">
+          Rollout History
+        </Typography>{" "}
+        {messages.slice(0, 1)}
+      </Stack>
+
+      <Collapse in={expanded} collapsedSize={0}>
```
Comment (shazlehu) [inline ui/src/components/RolloutHistory/RolloutHistory.tsx:80]: I think we want to restructure this like so:
```tsx
{data === null ? : (
    // Stack with circular progress
) : (
    // Collapse with messages & Expand button
)}
```

We don't need two null checks

Comment (shazlehu) [reply ui/src/components/RolloutHistory/RolloutHistory.tsx:80]: Also, button shouldn't appear when there's only one item in the history

File: ui/src/components/RolloutHistory/RolloutHistory.tsx (hunk 2 above)
Comment (Mrod1598) [reply ui/src/components/RolloutHistory/RolloutHistory.tsx]: Updated

File: ui/src/components/RolloutHistory/RolloutHistory.tsx (hunk 3 above)
Comment (Mrod1598) [reply ui/src/components/RolloutHistory/RolloutHistory.tsx]: Updated

File: ui/src/components/RolloutHistory/RolloutHistory.tsx [hunk 4]
```diff
@@ -54,30 +55,48 @@ export const RolloutHistory: React.FC<RolloutHistoryProps> = ({
...
+          {messages.slice(1)}
+        </Collapse>
       )}
 
-      {messages}
+      {
+        // only show the expand button if there are more than 1 messages
+        messages.length > 1 && (
+          <Stack direction="row" justifyContent="center" alignItems="center">
+            {" "}
```
Comment (shazlehu) [inline ui/src/components/RolloutHistory/RolloutHistory.tsx:92]:
```suggestion
```

File: ui/src/components/RolloutHistory/RolloutHistory.tsx [hunk 5]
```diff
@@ -105,7 +123,6 @@ function makeMessages(
         <>
           <Typography component="span"> with</Typography>
           <Typography color="error" component="span">
-            {" "}
```
Comment (dsvanlani) [inline ui/src/components/RolloutHistory/RolloutHistory.tsx:108]: We need this space here, otherwise

```
Version 4 rollout paused with1 error 
```

**Review by dsvanlani — APPROVED** [review]
Just include the space before merge.  Looks good.

**Review by shazlehu — APPROVED** [review]

**Pull request merged**